staging/src/k8s.io/apimachinery/pkg/util/wait/wait.go

```diff
@@ -379,6 +393,8 @@ func poller(interval, timeout time.Duration) WaitFunc {
 					case ch <- struct{}{}:
 					default:
 					}
+				case <-cancelCh:
+					return
```

Comment:
Sorry, I just now read the PR description. Shouldn't WaitForCacheSync just be fixed to close(done) instead? I don't see how this change helps, given that WaitForCacheSync still isn't calling the cancel function.

Reply:
I don't think users should know how to prevent `wait` package from leaking goroutines. 

Another way to fix this issue: the `WaitFor()` function creates a new channel and pass it to `WaitFunc`. Then close it after `WaitFor()` finishing.